app/components/form/SideModalForm.tsx

```diff
@@ -27,16 +36,16 @@ type SideModalFormProps<TFieldValues extends FieldValues> = {
    */
   children: ReactNode
   onDismiss: () => void
+  resourceName: string
   /** Must be provided with a reason describing why it's disabled */
   submitDisabled?: string
   /** Error from the API call */
   submitError: ApiError | null
   loading?: boolean
-  title: string
+  title?: string
```

Comment:
I’d add a doc comment here explaining that you only need this when you want to override the default create or edit title. Same with `submitLabel` I think.